flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeContext.java

```diff
@@ -113,6 +113,11 @@ public Integer getVersion() {
         return version;
     }
 
+    /** Returns a new {@code uid} for transformations. */
+    public String generateUid(String transformationName) {
+        return String.format("%s-%s-%s", getId(), getTypeAsString(), transformationName);
```

Comment:
the previous solution was fine, my comment was only referring to perform a regex check that `transformationName` is always lower case with dashes. we will also expose this method to users in one of the next PRs.